sympy/series/sequences.py

```diff
@@ -0,0 +1,522 @@
+from __future__ import print_function, division
+
+from sympy.core.expr import Expr
+from sympy.core.singleton import (S, Singleton)
+from sympy.core.compatibility import (range, integer_types, with_metaclass\
+                                      , is_sequence)
+from sympy.core.sympify import sympify
+from sympy.core.containers import Tuple
+from sympy.functions.elementary.integers import ceiling
+from sympy.utilities.misc import filldedent
+from sympy.sets.sets import Interval, Set
+
+
+class SeqBase(Expr):
+    """Base class for sequences"""
+
+    is_iterable = False
+
+    @property
+    def gen(self):
+        """Returns the generator for the sequence"""
+        return self._gen
+
+    @property
+    def _gen(self):
```

Comment:
I do see the reason you're implementing all the methods as public/private, tho I'm not sure overriding public methods from an abstract class is a bad thing. Also, in some cases it may be good to write new docstrings/doctests because the implementation of these properties varies slightly from one type of sequence to another. Since it is plain that this is a base class that shouldn't be used and anything you actually need will be implemented in subclasses, my vote would be to drop the `_foo` properties and move them to `foo`.